internal/mining/mining_harness_test.go

```diff
@@ -79,6 +79,54 @@ type fakeChain struct {
 	utxos                              *blockchain.UtxoViewpoint
 }
 
+// Returns a new fakeChain to be used for testing.
+// When using this function to create a fakeChain to pass to newMiningHarness(...), it is
+// recommended that the same chainParams used to call this function is used to call
+// newMiningHarness.
+// If the value of heightOfBestState is less than chainParams.CoinbaseMaturity, it will be
+// set to chainParams.CoinbaseMaturity
+func NewFakeChain(chainParams *chaincfg.Params, heightOfBestState int64) *fakeChain {
```

Comment:
It's a bit odd to take `heightOfBestState` as a parameter but don't allow setting anything else on the chain (including the other fields of best state).  It seems like it'd either be better to take a config type that holds everything or just let the caller set those fields on the chain as needed afterward (basically how it currently is).